src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,52 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * Represents the user interface that deals with interactions with the user.
+ */
+public class Ui {
+    /** Scanner to take in user input */
+    private final Scanner sc = new Scanner(System.in);
+
+    /**
+     * Public constructor.
+     */
+    public Ui() { }
```

Comment:
Not sure whether you really need this, Java has default constructors

Reply:
I included this for Javadoc purposes as stated [here](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#defaultconstructors), but I think I could include a longer comment and make sure this is consistent for other classes as well. Thank you for your comments!